### PR TITLE
feat: add squad pinned posts reordering

### PR DESCRIPTION
**Pull request by DragosIorgulescu**
API: https://github.com/dailydotdev/daily-api/pull/1496

## Changes

- adding support for & using the `swappedPinnedPosts` mutation

TODO:
- [x] use a separate mutation for swapping
- [x] agree on a long term solution

## Preview

pinned-post-reordering

## Events

No new events to track here, for now

### On those affected packages:
- [x] Have you done sanity checks in the webapp?
- [x] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [x] MobileL (420px)
- [x] Tablet (656px)
- [x] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1812 #done

### CODE REVIEW

File: packages/shared/src/components/Feed.tsx [hunk 1]
```diff
@@ -371,6 +371,8 @@ export default function Feed<T>({
       );
     },
     post,
+    prevPost: (items[postMenuIndex - 1] as PostItem)?.post,
+    nextPost: (items[postMenuIndex + 1] as PostItem)?.post,
```
Comment (rebelchris) [inline packages/shared/src/components/Feed.tsx:375]: Oef not against the implementation, but thinking about a leanFeed, this is yet another constraint in moving it.

File: packages/shared/src/graphql/posts.ts [hunk 2]
```diff
@@ -519,8 +519,18 @@ export const editPost = async (
 };
 
 export const PIN_POST_MUTATION = gql`
-  mutation UpdatePinPost($id: ID!, $pinned: Boolean!) {
-    updatePinPost(id: $id, pinned: $pinned) {
+  mutation UpdatePinPost(
+    $id: ID!
+    $pinned: Boolean!
+    $nextPostId: ID
+    $prevPostId: ID
```
Comment (rebelchris) [inline packages/shared/src/graphql/posts.ts:526]: Do we need this change?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Nice work, very minor questions otherwise ok.

**Review by sshanzel — COMMENTED** [review]
Great work. Nothing blocking but we should maybe include the new icons on the index file of the folder itself.

File: packages/shared/src/graphql/posts.ts (hunk 2 above)
Comment (DragosIorgulescu) [reply packages/shared/src/graphql/posts.ts]: good catch, removed ✂️

**Comment by DragosIorgulescu** [comment]
> Great work. Nothing blocking but we should maybe include the new icons on the index file of the folder itself.

@sshanzel good point, however, does tree shaking drop any unused imports if exported the way these are in [index](https://github.com/dailydotdev/apps/blob/f5191754485e855f295539d2e1797672c88e19ca/packages/shared/src/components/icons/index.ts)?

**Comment by DragosIorgulescu** [comment]
closing this here as I manually squash merged into main

**Pull request closed without merging**